src/server/scene/basic_text_input_hub.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright © 2021 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored By: William Wold <william.wold@canonical.com>
+ */
+
+#ifndef MIR_SCENE_BASIC_TEXT_INPUT_HUB_H_
+#define MIR_SCENE_BASIC_TEXT_INPUT_HUB_H_
+
+#include "mir/scene/text_input_hub.h"
+#include "mir/observer_multiplexer.h"
+
+#include <mutex>
+
+namespace mir
+{
+namespace scene
+{
+class TextInputStateObserverMultiplexer: mir::Executor, public ObserverMultiplexer<TextInputStateObserver>
```

Comment:
Heh. We could really do with a `mir::system_executor`, couldn't we ;)

Reply:
Or have `ObserverMultiplexer` implement `Executor`, so we don't have to implement it in each subclass.

But this is not the time to decide on that.